test.js

```diff
@@ -37,3 +37,33 @@ describe('versioned', function () {
     })
   })
 })
+
+describe('url parse', function () {
+  var builtinUrlParse = require('url').parse
+
+  it('handles https:// url', function () {
+    var url = 'https://foo.com/bar'
+    var parsed = builtinUrlParse(url)
```

Comment:
I needed to check if the built-in parsing handles `something@else.com` form, seems it does not